README.md

```diff
@@ -145,7 +145,7 @@ The process is the same for encoding, except:
  * has been specified as an ENTRY_TYPE in the cmake call. */
 Pet_t pet = { /* Initialize with desired data. */ };
 uint8_t output[100]; /* 100 is an example. Must be large enough for data to fit. */
-size_t out_len;
+uint32_t out_len;
```

Comment:
Why not `int` all over the place? ;)

Reply:
Heresy!

Reply:
:)

We might want to use `unsigned int` eventually, but that would take some more consideration about the particulars of encoding/decoding integers of different lengths.

Reply:
According to Bjarne the man:

```
Use int until you have a reason not to.
Use unsigned only for bit patterns.
Never mix signed and unsigned
```

Reply:
I see now that you are explaining the explicit 32 bit limit in your code, forget this then :)